sdk/eventhub/Azure.Messaging.EventHubs/assets/identity-samples-azure-setup.ps1

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+  .SYNOPSIS
+    Performs the tasks needed to setup a service principal to perform authentication against 
+    Azure Active Directory and sets the roles needed to access Event Hubs resources.
+
+    It creates an Azure Event Hub Namespace and an Azure Event Hub using the specified names.
+
+  .DESCRIPTION
+    This script handles creation of a service principal and assigns the role 
+    "Azure Event Hubs Data Owner" to the resource group whose name is passed in.
+
+    The script attempts creation of an Azure Event Hubs Namespace and an Azure Event Hub
+    using the names passed in as arguments.
+    
+    Upon completion, the script will output the principal information.
+ 
+    For more detailed help, please use the -Help switch. 
+#>
+
+# =======================
+# == Script Parameters ==
+# =======================
+
+[CmdletBinding(DefaultParameterSetName="Help")] 
+[OutputType([String])] 
+param
+( 
+  [Parameter(Mandatory=$true, ParameterSetName="Help", Position=0)]
+  [Switch] $Help,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute", Position=0)]
+  [ValidateNotNullOrEmpty()]
+  [string] $SubscriptionName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $ResourceGroupName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $NamespaceName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [string] $EventHubName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [ValidateNotNullOrEmpty()]
+  [ValidateScript({ $_.Length -ge 6})]
+  [string] $ServicePrincipalName,
+
+  [Parameter(Mandatory=$true, ParameterSetName="Execute")]
+  [AllowNull()]  
+  [string] $AzureRegion = $null
+)
+
+# =====================
+# == Module Imports  ==
+# =====================
+
+Import-Module Az.Resources
+
+# ==========================
+# == Function Definitions ==
+# ==========================
+
+. .\functions\azure-principal-functions.ps1
+
+function DisplayHelp
+{
+  <#
+    .SYNOPSIS
+      Displays the usage help text.
+
+    .DESCRIPTION
+      Displays the help text for usage.
+
+    .OUTPUTS
+      Help text for usage to the console window.
+  #>
+
+  $indent = "    "
+
+  Write-Host "`n"
+  Write-Host "Creation of an Azure Active Directory Service Principal"
```

Comment:
We may want to consider tweaking this since we're now pretty much building up resources to support identity samples all-up.